plugins/outputs/signalfx/signalfx.go

```diff
@@ -0,0 +1,238 @@
+package signalfx
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/signalfx/golib/v3/datapoint"
+	"github.com/signalfx/golib/v3/datapoint/dpsink"
+	"github.com/signalfx/golib/v3/event"
+	"github.com/signalfx/golib/v3/sfxclient"
+	"github.com/signalfx/signalfx-go/realm"
+)
+
+/*init initializes the plugin context*/
+func init() {
+	outputs.Add("signalfx", func() telegraf.Output {
+		return NewSignalFx()
+	})
+}
+
+// SignalFx plugin context
+type SignalFx struct {
+	AccessToken        string   `toml:"access_token"`
+	SignalFxRealm      string   `toml:"signalfx_realm"`
+	IngestURL          string   `toml:"ingest_url"`
+	IncludedEventNames []string `toml:"included_event_names"`
+
+	includedEventSet map[string]bool
+	client           dpsink.Sink
+
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+}
+
+var sampleConfig = `
+    ## SignalFx Org Access Token
+    access_token = "my-secret-token"
+
+    ## The SignalFx realm that your organization resides in
+	signalfx_realm = us9  # Required if ingest_url is not set
+
+	## You can optionally provide a custom ingest url instead of the
+	## signalfx_realm option above if you are using a gateway or proxy
+	## instance.  This option takes precident over signalfx_realm.
+	ingest_url = https://my-custom-ingest/
+
+    ## Event typed metrics are omitted by default,
+    ## If you require an event typed metric you must specify the
+    ## metric name in the following list.
+    included_event_names = ["plugin.metric_name", ""]
+`
+
+// GetMetricType returns the equivalent telegraf ValueType for a signalfx metric type
+func GetMetricType(mtype telegraf.ValueType) (metricType datapoint.MetricType) {
+	switch mtype {
+	case telegraf.Counter:
+		metricType = datapoint.Counter
+	case telegraf.Gauge:
+		metricType = datapoint.Gauge
+	case telegraf.Summary:
+		metricType = datapoint.Gauge
+	case telegraf.Histogram:
+		metricType = datapoint.Gauge
+	case telegraf.Untyped:
+		metricType = datapoint.Gauge
+	default:
+		metricType = datapoint.Gauge
+	}
+	return metricType
+}
+
+// NewSignalFx - returns a new context for the SignalFx output plugin
+func NewSignalFx() *SignalFx {
+
+	ctx, cancel := context.WithCancel(context.Background())
+	return &SignalFx{
+		AccessToken:        "",
+		SignalFxRealm:      "",
+		IngestURL:          "",
+		IncludedEventNames: []string{""},
+		ctx:                ctx,
+		cancel:             cancel,
+	}
+}
+
+/*Description returns a description for the plugin*/
```

Comment:
Use `//` and `/**/` consistently (prefer the former, it seems better aligned with the rest of the codebase)

Reply:
Fixed.